matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/model/create/CreateRoomParams.kt

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.matrix.android.api.session.room.model.create
+
+import android.util.Patterns
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import im.vector.matrix.android.api.MatrixPatterns.isUserId
+import im.vector.matrix.android.api.auth.data.Credentials
+import im.vector.matrix.android.api.auth.data.HomeServerConnectionConfig
+import im.vector.matrix.android.api.session.events.model.Event
+import im.vector.matrix.android.api.session.events.model.EventType
+import im.vector.matrix.android.api.session.events.model.toContent
+import im.vector.matrix.android.api.session.room.model.PowerLevels
+import im.vector.matrix.android.api.session.room.model.RoomDirectoryVisibility
+import im.vector.matrix.android.api.session.room.model.RoomHistoryVisibility
+import im.vector.matrix.android.internal.auth.data.ThreePidMedium
+import java.util.*
+
+/**
+ * Parameter to create a room, with facilities functions to configure it
+ */
+@JsonClass(generateAdapter = true)
+class CreateRoomParams {
+
+    /**
+     * A public visibility indicates that the room will be shown in the published room list.
+     * A private visibility will hide the room from the published room list.
+     * Rooms default to private visibility if this key is not included.
+     * NB: This should not be confused with join_rules which also uses the word public. One of: ["public", "private"]
+     */
+    var visibility: RoomDirectoryVisibility? = null
```

Comment:
Better if we can get val and not nullable by using some kind of factory, instead of updating it.